### ISSUE TITLE
yii\rest\Serializer serialize ArrayDataProvider error

### ISSUE THREAD

**Issue by dcb9** [yiisoft/yii2#11977]
#### Creating a Controller
First, create a controller class app\controllers\UserController as follows,

```
namespace app\controllers;

use yii\data\ArrayDataProvider;
use yii\rest\Controller;

class UserController extends Controller
{
    public function actionIndex()
    {
        $users = [
            ['id' => 1, 'nickname' => 'Bob'],
            ['id' => 2, 'nickname' => 'Tom'],
        ];

        return new ArrayDataProvider([
            'allModels' => $users,
            'pagination' => [
                'pageSize' => 1,
            ]
        ]);
    }
}
```

#### Configuring URL Rules
```
'urlManager' => [
    'enablePrettyUrl' => true,
    'enableStrictParsing' => true,
    'showScriptName' => false,
    'rules' => [
        ['class' => 'yii\rest\UrlRule', 'controller' => 'user'],
    ],
]
```

#### Serve

`$ php yii serve`

in the other terminal:

`$ curl http://localhost:8080/users?page=2`

### What is the expected result?

```
[{"id":2,"nickname":"Tom"}]
```

### What do you get instead?

```
{"1":{"id":2,"nickname":"Tom"}}
```

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.10
| PHP version      | 5.5.34
| Operating system | Mac OS X

**Comment by klimov-paul** [comment]
This is expected behavior.

**Comment by dcb9** [comment]
And what do you want to do with this?

**Comment by klimov-paul** [comment]
Nothing: you are requesting the list of models via REST (second page to be precise):

> $ curl http://localhost:8080/users?page=2

So you get the list of models in return (second page to be precise).

Everything is correct.

**Comment by Forin** [comment]
The question is why "1:" is inserted when the ?page=2 parameter is added.
And why does the response-format differ:
?page=1:` [{"id":1,"nickname":"Bob"}]`
?page=2:` {"1":{"id":2,"nickname":"Tom"}}`

**Comment by klimov-paul** [comment]
I see now.
It is caused by `ArrayDataProvider` pagination mechanism, which splits original array preserving its keys.

**Comment by dcb9** [comment]
➜  ~ cat a.php
```
<?php

	$arr = [ ['id' => 1, 'nickname' => 'Bob'] ];

	echo json_encode($arr), PHP_EOL;
	$arr = [ 1 => ['id' => 2, 'nickname' => 'Tom'] ];

	echo json_encode($arr), PHP_EOL;
```
➜  ~ php a.php
```
[{"id":1,"nickname":"Bob"}]
{"1":{"id":2,"nickname":"Tom"}}
```

**Comment by fernandezekiel** [comment]
interesting, maybe this is more of a fix for ther serializer?

**Comment by klimov-paul** [comment]
Resolved by commit cd8c01f555e425d4654775002ae5f44152e0c4b1
